Cargo.toml

```diff
@@ -94,6 +94,7 @@ enum-as-inner = "0.6"
 idna = "0.5"
 ipconfig = "0.3.0"
 ipnet = "2.3.0"
+libc = "0.2"
```

Comment:
Did you consider using nix or rustix instead?

Reply:
Rustix won't work because it does not expose setuid or related functions.  I considered using something like nix, but:

* Nix is another dependency on top of libc
* It has narrower OS support than libc
* The unsafe libc functions here are pretty simple in terms of API and I think easy to reason about the safety of their use
* If this somehow blows up, it will cause a failure on startup. It isn't something that will ever get called or get exposed in-operation or in response to network transactions.  IOW, the risk of failure seems very low.

I'm not totally opposed to nix, but it doesn't seem worth it for this function.